pass.go

```diff
@@ -15,7 +15,7 @@ type Pass interface {
 	// VisitJobPost is callback when visiting Job node after visiting its children. It returns internal error when it cannot continue the process
 	VisitJobPost(node *Job) error
 	// VisitWorkflowPre is callback when visiting Workflow node before visiting its children. It returns internal error when it cannot continue the process
-	VisitWorkflowPre(node *Workflow) error
+	VisitWorkflowPre(path string, node *Workflow) error
```

Comment:
I need the `path` here to infer the name from the filename, because it's not always defined explicitly on the workflow's YAML.

I also explored adding a new `VisitFile` method to avoid touching many failed to match the new `VisitWorkflowPre` signature, but because the stateful nature of this rule, and the concurrent approach for processing files, the current name stored as a field of the rule itself gets overridden, leading to a race condition.

So I end up adding this extra parameter here to compute the final name.

I could also add the final name to the `Workflow` structure, but it doesn't feel right, as that stands for a pure YAML model.